superduperdb/components/graph.py

```diff
@@ -0,0 +1,220 @@
+import dataclasses as dc
+import typing as t
+from collections import OrderedDict
+
+import networkx as nx
+
+from superduperdb.components.component import Component
+from superduperdb.components.model import Model, _Predictor
+
+
+class GraphModel:
+    def __init__(self, name):
+        self.identifier = name
+        self.inputs = []
+
+    def predict(self, x):
+        return x
+
+
+class IntermediateGraphNode:
+    def __init__(self, node, graph):
+        self.node = node
+        self.G = graph
+
+    @property
+    def output(self):
+        return self.G.get_output(self.node)
+
+
+@dc.dataclass(kw_only=True)
+class Graph(Component, _Predictor):
+    models: t.List[Model] = dc.field(default_factory=list)
```

Comment:
`_Predictor`

Reply:
`t.List[_Predictor]`

Reply:
Can use _Predictor, since It does not have identifier ?

Reply:
It has in the latest `main`.